lib/errorlogger.cpp

```diff
@@ -945,24 +945,40 @@ std::string getClassification(const std::string &guideline, ReportType reportTyp
         return getClassification(checkers::certCInfo, guideline);
     case ReportType::certCpp:
         return getClassification(checkers::certCppInfo, guideline);
-    case ReportType::misraC:
+    case ReportType::misraC2012:
+    case ReportType::misraC2023:
+    case ReportType::misraC2025:
     {
-        auto components = splitString(guideline, '.');
+        const bool isDirective = guideline.find("Dir ") == 0;
```

Comment:
this is pretty inefficient in general. but well in this case the strings are really short, 4-5 characters max.. so I can allow it. but I am thinking that maybe it's better to write `guideline.rfind("Dir ", 0)` to only check 1 position..